pkg/controller/scylladbdatacenter/resource.go

```diff
@@ -316,22 +317,21 @@ func StatefulSetForRack(rack scyllav1alpha1.RackSpec, sdc *scyllav1alpha1.Scylla
 	requiredLabels[naming.ScyllaVersionLabel] = scyllaDBVersion
 	maps.Copy(requiredLabels, selectorLabels)
 
+	sdcLabels := makePropagableLabels(sdc)
+
 	rackLabels := map[string]string{}
-	maps.Copy(rackLabels, sdc.Labels)
 	maps.Copy(rackLabels, requiredLabels)
+	maps.Copy(rackLabels, sdcLabels)
```

Comment:
sdc.Labels and requiredLabels were swapped, which may be incorrect, as required ones should be added at the end to win potential collisions. Any reason why this is changed?